pkg/network/protocols/http/event_pipeline_test.go

```diff
@@ -0,0 +1,173 @@
+	mockBatch := events.Batch{
+		Len:        uint16(numOfEventsInBatch),
+		Cap:        uint16(numOfEventsInBatch),
+		Event_size: uint16(unsafe.Sizeof(httpEvents[0])),
+		Data:       eBPFEventToBytes(b, httpEvents, numOfEventsInBatch),
+	}
+
+	for i := 0; i < totalEvents/numOfEventsInBatch; i++ {
+		mockBatch.Idx = uint64(i)
+		var buf bytes.Buffer
+		require.NoError(b, binary.Write(&buf, binary.LittleEndian, &mockBatch))
+		events.RecordSample(c, consumer, buf.Bytes())
+		buf.Reset()
+	}
+}
+
+// createHTTPEvents creates a slice of HTTP events to be used in the benchmark.
+func createHTTPEvents(eventsData []HTTPEventData) []EbpfEvent {
+	events := make([]EbpfEvent, len(eventsData))
+	for i, data := range eventsData {
+		events[i] = EbpfEvent{
+			Tuple: ConnTuple{},
+			Http: EbpfTx{
+				Request_started:      1,
+				Response_last_seen:   2,
+				Request_method:       data.Method,
+				Response_status_code: data.StatusCode,
+				Request_fragment:     requestFragment(data.RequestFragment),
+			},
+		}
+	}
+	return events
+}
+
+// BenchmarkHTTPEventConsumer benchmarks the consumer with a large number of events to measure the performance.
+func BenchmarkHTTPEventConsumer(b *testing.B) {
+	// Set MemProfileRate to 1 in order to collect every allocation
+	runtime.MemProfileRate = 1
+	var wg sync.WaitGroup
+
+	b.ReportAllocs()
+	b.ResetTimer()
+
+	testCases := []struct {
+		name             string
+		totalEventsCount int
+		// Serialized data can't exceed batchDataSize bytes that why we can insert 14 events in a batch.
+		numOfEventsInBatch int
+		httpEvents         []EbpfEvent
+	}{
+		{"SmallBatch", 1000, 8, createHTTPEvents([]HTTPEventData{
+			{Method: uint8(MethodGet), StatusCode: http.StatusOK, RequestFragment: []byte("GET / HTTP/1.1")},
+			{Method: uint8(MethodPost), StatusCode: http.StatusCreated, RequestFragment: []byte("POST /create HTTP/1.1")},
+		})},
+		{"MediumBatch", 38000, 10, createHTTPEvents([]HTTPEventData{
+			{Method: uint8(MethodGet), StatusCode: http.StatusOK, RequestFragment: []byte("GET / HTTP/1.1")},
+			{Method: uint8(MethodPost), StatusCode: http.StatusCreated, RequestFragment: []byte("POST /create HTTP/1.1")},
+		})},
+		{"LargeBatch", 42000, 14, createHTTPEvents([]HTTPEventData{
+			{Method: uint8(MethodGet), StatusCode: http.StatusOK, RequestFragment: []byte("GET / HTTP/1.1")},
+			{Method: uint8(MethodPost), StatusCode: http.StatusCreated, RequestFragment: []byte("POST /create HTTP/1.1")},
+		})},
+		{"MaxEventsCount", 3150000, 14, createHTTPEvents([]HTTPEventData{
+			{Method: uint8(MethodGet), StatusCode: http.StatusOK, RequestFragment: []byte("GET / HTTP/1.1")},
+			{Method: uint8(MethodDelete), StatusCode: http.StatusAccepted, RequestFragment: []byte("DELETE /delete HTTP/1.1")},
+		})},
+	}
+
+	for _, tc := range testCases {
+		b.Run(tc.name, func(b *testing.B) {
+			for i := 0; i < b.N; i++ {
+				consumer, p := setupBenchmark(b, config.New(), tc.totalEventsCount, tc.numOfEventsInBatch, tc.httpEvents, &wg)
```

Comment:
You don't wait on the wait group. Also, consider returning it from setupBenchmark instead of initialising it in the outer scope (to avoid the use of an output parameter).

Reply:
I added the wait. Regarding why I am using the wg, I am creating it outside the function and passing it as a parameter to ensure the wait applies only to generateMockEvents.

Reply:
I believe you can still create it within the function. Nevertheless, this is a minor detail and not blocking approval.